master/buildbot/process/buildrequest.py

```diff
@@ -267,7 +267,17 @@ def canBeCollapsed(master, br1, br2):
         if br1['buildsetid'] == br2['buildsetid']:
             defer.returnValue(True)
 
-        # get the buidlsets for each buildrequest
+        # Merge requests should only be collapsed with other
```

Comment:
I think this fails testing because when the new buildset is not a merge request it will not run the new logic and will collapse with non merge requests.

I think the following changes might make sense (they fix the test for me at least)? :
```patch
diff --git a/master/buildbot/process/buildrequest.py b/master/buildbot/process/buildrequest.py
index a6f83ce9f..2923d49e8 100644
--- a/master/buildbot/process/buildrequest.py
+++ b/master/buildbot/process/buildrequest.py
@@ -270,8 +270,12 @@ class BuildRequest(object):
         # Merge requests should only be collapsed with other
         # merge requests from the same branch of the same source project
         bsp1 = yield master.db.buildsets.getBuildsetProperties(br1['buildsetid'])
+        bsp2 = yield master.db.buildsets.getBuildsetProperties(br2['buildsetid'])
+
+        if bsp1.get('event') != bsp2.get('event'):
+            defer.returnValue(False)
+
         if bsp1.get('event') == ('merge_request', 'Change'):
-            bsp2 = yield master.db.buildsets.getBuildsetProperties(br2['buildsetid'])
             if (bsp2.get('event') != ('merge_request', 'Change') or
                     bsp1.get('source_branch') != bsp2.get('source_branch') or
                     bsp1.get('source_git_ssh_url') != bsp2.get('source_git_ssh_url')):
```